frontend/packages/pipelines-plugin/src/components/pipelineruns/logs/Logs.tsx

```diff
@@ -34,17 +35,28 @@ const Logs: React.FC<LogsProps> = ({
   const [error, setError] = React.useState<boolean>(false);
   const resourceStatusRef = React.useRef<string>(resourceStatus);
   const onCompleteRef = React.useRef<(name) => void>();
+  const blockContentRef = React.useRef<string>('');
   onCompleteRef.current = onComplete;
 
+  const addContentAndScroll = throttle(() => {
+    if (contentRef.current) {
+      contentRef.current.innerText += blockContentRef.current;
+    }
+    if (scrollToRef.current) {
+      scrollToRef.current.scrollIntoView({ behavior: 'smooth', block: 'end' });
+    }
+    blockContentRef.current = '';
+  }, 300);
+
+  const contentScrollerRef = React.useRef<() => void>(addContentAndScroll);
```

Comment:
Why do we need a ref for a throttled function?

Reply:
Ref is not needed and I have updated the code to use useCallback.